cms/taxonomy/management/commands/sync_topics.py

```diff
@@ -0,0 +1,210 @@
+        )
+    ]
+
+
+def _sync_with_fetched_topics(fetched_topics: Iterable[Mapping[str, str]]) -> None:
+    """For each fetched topic, decide if it needs to be created, updated, or left as is."""
+    _check_for_duplicate_topics(fetched_topics)
+
+    updated_count = 0
+    created_count = 0
+    for fetched_topic in fetched_topics:
+        if existing_topic := _get_topic(fetched_topic["id"]):
+            if not _topic_matches(existing_topic, fetched_topic):
+                _update_topic(existing_topic, fetched_topic)
+                updated_count += 1
+        else:
+            _create_topic(fetched_topic)
+            created_count += 1
+
+    logger.info("Saved %d new topic(s)", created_count)
+    logger.info("Updated %d existing topic(s)", updated_count)
+
+
+def _check_for_duplicate_topics(fetched_topics: Iterable[Mapping[str, str]]) -> None:
+    topic_ids = set()
+    for topic in fetched_topics:
+        if topic["id"] in topic_ids:
+            raise RuntimeError(f"Received duplicate topic ID in API responses, topic IDs must be unique: {topic['id']}")
+        topic_ids.add(topic["id"])
+
+
+def _get_topic(topic_id: str) -> Topic | None:
+    """Fetches a Topic record by its primary key (id), or returns None if not found."""
+    return Topic.objects.filter(id=topic_id).first()
+
+
+def _topic_matches(existing_topic: Topic, fetched_topic: Mapping[str, str]) -> bool:
+    """Compares: title, description, removed status, Parent ID.
+    If all these match, the function returns True; otherwise False.
+    """
+    existing_parent = existing_topic.get_parent()
+    existing_parent_id = existing_parent.id if existing_parent else None
+
+    return (
+        fetched_topic["title"] == existing_topic.title
+        and fetched_topic.get("description") == existing_topic.description
+        and not existing_topic.removed
+        and existing_parent_id == fetched_topic.get("parent_id")
+    )
+
+
+def _update_topic(existing_topic: Topic, fetched_topic: Mapping[str, str]) -> None:
+    """Updates an existing topic with data from an external source.
+
+    - Updates `title`, `description`, and sets `removed = False` (as it is present in external data).
+    - If the `parent_id` differs from the current parent, moves the topic to the new parent.
+    - Logs changes and warnings when moving topics.
+    """
+    logger.info("Updating existing topic: %s with fetched topic: %s", existing_topic.id, fetched_topic)
+    existing_topic.title = fetched_topic.get("title")
+    existing_topic.description = fetched_topic.get("description")
+    existing_topic.removed = False
+    existing_topic.save()
+
+    # If the topic parent has changed then move the node to match
+    existing_parent_id: str | None = getattr(existing_topic.get_parent(), "id", None)
+    if existing_parent_id != fetched_topic.get("parent_id"):
+        parent_id = fetched_topic.get("parent_id")
+        new_parent = _get_topic(parent_id) if parent_id else None
```

Comment:
```suggestion
    # If the topic parent has changed then move the node to match
    existing_parent_id: str | None = getattr(existing_topic.get_parent(), "id", None)
    new_parent_id: str = fetched_topic.get("parent_id")
    if existing_parent_id != new_parent_id:
        new_parent = _get_topic(new_parent_id) if parent_id else None
```
Super minor but imho makes it read better

Reply:
`new_parent_id: str` > `new_parent_id: str | None`. Since it is using `.get` the value can be `None`.